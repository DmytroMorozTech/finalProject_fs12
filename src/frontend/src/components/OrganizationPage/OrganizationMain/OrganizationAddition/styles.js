import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  item: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    cursor: 'pointer',
    color: theme.palette.grey[700],
    // zIndex: '10',
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.1)'
    }
  },

  icon: {
    fontSize: theme.typography.icons.medium.fontSize,
    marginRight: theme.spacing(2),
    display: 'flex'
  },

  title: {
    display: 'flex',
    fontSize: theme.typography.h5.fontSize,
    fontWeight: theme.typography.fontWeightBold
  }
}))
