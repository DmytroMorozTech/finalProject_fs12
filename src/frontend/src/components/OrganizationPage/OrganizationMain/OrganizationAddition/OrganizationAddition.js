import { MenuItem } from '@material-ui/core'
import React from 'react'
import styles from './styles'

function OrganizationAddition ({ icon, title, onClick }) {
  const classes = styles()

  return (
    <MenuItem className={classes.item} onClick={onClick}>
      <div className={classes.icon}>{icon}</div>
      <div className={classes.title}>{title}</div>
    </MenuItem>
  )
}

export default OrganizationAddition
