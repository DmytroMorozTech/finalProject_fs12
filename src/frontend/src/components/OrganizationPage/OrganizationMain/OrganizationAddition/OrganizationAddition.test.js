import { shallow } from 'enzyme'
import React from 'react'
import OrganizationAddition from './OrganizationAddition'
import TelegramIcon from '@material-ui/icons/Telegram'

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<OrganizationAddition {...props}/>)

describe('should render OrganizationAddition component', () => {
  let icon
  let title
  let onClick

  let component
  beforeEach(() => {
    icon = <TelegramIcon fontSize="inherit"/>
    title = 'Share in a message'
    onClick = jest.fn()

    component = setUp({ icon, title, onClick })
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
