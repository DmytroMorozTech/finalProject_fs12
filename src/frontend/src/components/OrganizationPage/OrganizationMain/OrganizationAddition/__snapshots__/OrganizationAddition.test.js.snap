// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OrganizationAddition component to match snapshot 1`] = `
<WithStyles(ForwardRef(MenuItem))
  onClick={[MockFunction]}
>
  <div>
    <Memo(ForwardRef(TelegramIcon))
      fontSize="inherit"
    />
  </div>
  <div>
    Share in a message
  </div>
</WithStyles(ForwardRef(MenuItem))>
`;
